examples/js/loaders/GLTFLoader.js

```diff
@@ -1153,103 +1153,118 @@ THREE.GLTFLoader = ( function () {
 	function addMorphTargets( mesh, meshDef, primitiveDef, accessors ) {
 
 		var geometry = mesh.geometry;
-		var material = mesh.material;
-
 		var targets = primitiveDef.targets;
-		var morphAttributes = geometry.morphAttributes;
 
-		morphAttributes.position = [];
-		morphAttributes.normal = [];
+		var hasMorphPosition = false;
+		var hasMorphNormal = false;
+
+		for ( var i = 0, il = targets.length; i < il; i ++ ) {
+
+			var target = targets[ i ];
+
+			if ( target.POSITION !== undefined ) hasMorphPosition = true;
+			if ( target.NORMAL !== undefined ) hasMorphNormal = true;
 
-		material.morphTargets = true;
+			if ( hasMorphPosition && hasMorphNormal ) break;
+
+		}
+
+		// unnecessary to set morph if base attributes don't exist
+		if ( geometry.attributes.position === undefined ) hasMorphPosition = false;
+		if ( geometry.attributes.normal === undefined ) hasMorphNormal = false;
```

Comment:
From the spec: 

> A Morph Target is a morphable Mesh where primitives' attributes are obtained by adding the original attributes to a weighted sum of targets attributes.

My reading of that would be, base attributes MUST be set for any attributes used with morph targets. I don't think we need to ignore morph targets gracefully in this case, it would be fine to have a runtime error and enforce the spec strictly.

Reply:
Agreed with that. Updated, thanks.

BTW, even if base attribute isn't set but morph attribute is set, Three.js doesn't seem to show any errors/warnings. Anyways, users could be aware of something wrong in glTF because of unexpected rendering.